iothub/device/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs

```diff
@@ -72,5 +76,10 @@ public int GetHashCode(SharedAccessSignatureAuthorizationRule rule)
 
             return hashKeyName ^ hashPrimaryKey ^ hashSecondaryKey ^ hashRights;
         }
+
+        public override int GetHashCode()
```

Comment:
What exactly does this do?

Reply:
Directly from the .NET documentation:
Serves as the default hash function.

for complex objects, you can override how the hash code is generated.
All Equatable objects require this method to be implemented by the class.

Reply:
I know what GetHashCode is and why it exists, I didn't understand what calling the same method with `this` as a parameter did, but I saw later in another file it is delegated to another (now private) method.

Why not just put that implementation in here?